boards/DCM/Src/App/states/App_DriveState.c

```diff
@@ -27,16 +33,28 @@ static void DriveStateRunOnTick100Hz(struct StateMachine *const state_machine)
     App_AllStatesRunOnTick100Hz(state_machine);
 
     struct DcmWorld *world = App_SharedStateMachine_GetWorld(state_machine);
-    struct DcmCanRxInterface *can_rx = App_DcmWorld_GetCanRx(world);
+    struct DcmCanRxInterface *can_rx      = App_DcmWorld_GetCanRx(world);
+    struct ErrorTable *       error_table = App_DcmWorld_GetErrorTable(world);
+    const struct State *      next_state  = App_GetDriveState();
 
-    if (App_CanRx_DIM_SWITCHES_GetSignal_START_SWITCH(can_rx) ==
-        CANMSGS_DIM_SWITCHES_START_SWITCH_OFF_CHOICE)
-    {
-        App_SharedStateMachine_SetNextState(state_machine, App_GetInitState());
-    }
+    const bool has_inverter_faulted = HasInverterFaulted(can_rx);
+    const bool is_critical_error_present =
+        App_SharedErrorTable_HasAnyCriticalErrorSet(error_table);
+    const bool is_start_switch_off = IsStartSwitchOff(can_rx);
```

Comment:
we dont need these const bools anymore, can just call the functions in the if-else statements directly

Reply:
fixed